### PR TITLE
Add support for passing a block to `start` method

### PR DESCRIPTION
**Pull request by GustavoCaso**
Add a new module function `execute(filename)` into `GvlTracing` wrapping call of `start` and `stop` methods.

This is one of the tasks from https://github.com/ivoanjo/gvl-tracing/issues/13

### CODE REVIEW

**Comment by GustavoCaso** [comment]
If we do not like the `execute` method we could rename the C methods to `_start` and `_stop` and add simply wrapper in Ruby land. 

```ruby
module GvlTracing
  class << self

    private :_start
    private :_stop

    def start(filename)
      _start(filename)

      if block_given?
        yield
        _stop
      end
    end

    def stop
      _stop
    end
  end
end
```

**Comment by ivoanjo** [comment]
Looks good! In #12 indeed we already are wrapping `start`/`stop` with Ruby versions as you suggested -- do you mind rebasing on top of master and adding the `block_given?` to `start` directly as you suggested?

File: lib/gvl-tracing.rb [hunk 1]
```diff
@@ -37,6 +37,11 @@ class << self
     def start(file)
       _start(file)
       @path = file
+
+      if block_given?
+        yield
```
Comment (GustavoCaso) [inline lib/gvl-tracing.rb:42]: @ivoanjo we might wan to think about error handling 😄

Comment (GustavoCaso) [reply lib/gvl-tracing.rb:44]: Done [53a21ca](https://github.com/ivoanjo/gvl-tracing/pull/14/commits/53a21ca9c3fdabd297ae19fb0fcb3237b10ac76e)

**Review by ivoanjo — APPROVED** [review]
Awesome, thanks :+1:

**Comment by ivoanjo** [comment]
Thanks @GustavoCaso for the contribution! I've (finally, lol, thanks for the patience!) released it as [v1.3.0](https://github.com/ivoanjo/gvl-tracing/releases/tag/v1.3.0) :)

**Pull request merged**